lightwood/model/neural.py

```diff
@@ -267,10 +269,12 @@ def fit(self, ds_arr: List[EncodedDs]) -> None:
                     ConcatedEncodedDs(train_ds_arr[subset_idx * 9: (subset_idx + 1) * 9]),
                     batch_size=200, shuffle=True)
 
-                stop_after = self.stop_after / 4
+                stop_after = self.stop_after / 8 if not self.timeseries_settings.is_timeseries else self.stop_after
+                return_model_after = (20000 if subset_itt > 0 else 1) if not self.timeseries_settings.is_timeseries \
```

Comment:
Isn't this `20000 if subset_it  or self.timeseries_settings.is_timeseries else 1` ?

Also... why do this for time series? The idea in general is to do a "warmup" on each subset (1 epoch than return the model) *then* train until completion (or until ~1/3rd of the time runs out) on each subset.

Reply:
> Isn't this 20000 if subset_it or self.timeseries_settings.is_timeseries else 1 ?

Fixed

> Also... why do this for time series? The idea in general is to do a "warmup" on each subset (1 epoch than return the model) then train until completion (or until ~1/3rd of the time runs out) on each subset.

Yes, but this scheme leads to underfitting in seemingly all TS datasets. I’m not sure why, but the issue with current approach seems to be as follows in e.g. monthly sunspots dataset:

* warmup round is quite short -as expected- and loss barely decreases
* in the “full” training round (i.e. subset_itt=1) model loss is stuck, which early stops training based on the error mean being 0.0
* forecasts are bad

If, instead, I crank epochs to 20k in all rounds (as in this PR):
* warmup round does a nice amount of epochs (120 approx.) and early stops correctly but, crucially, the loss this time around did decrease a lot
* “full” round once again is not able to decrease loss by more than a marginal amount (this time, loss is not constant), gets early stopped after 5 epochs
* predictor gives good forecasts

The weird thing to me is that we are sharing optimizer between max_fit calls, and LR does not change, so why would the second loop have stiffer learning dynamics if the DL has shuffle=True, I do not know… :thinking_face:

Reply:
Okay, as discussed: warmup has been scrapped in favour of simplicity. Let's see what the automated benchmarks think of this.